### PR TITLE
Add attribute for _disableAccessibilityState

### PR DESCRIPTION
**Pull request by chris-steele**
issue https://github.com/adaptlearning/adapt_framework/issues/1795

Update rather than recreate component state and added `_disableAccessbilityState` option for components

### CODE REVIEW

File: src/core/js/views/componentView.js [hunk 1]
```diff
@@ -25,24 +25,18 @@ define([
...
+            var $rendered = $(Handlebars.partials['state']( _.extend(this.model.toJSON(), {a11yConfig:Adapt.config.get('_accessibility')}) ));
 
-            //restore previous tab index if not on
-            var previousTabIndex = $previousState.find(".aria-label").attr("tabindex");
-            if (previousTabIndex == "-1") {
-                $rendered.find(".aria-label").attr("tabindex", previousTabIndex);
-            }
-
-            this.$el.append( $rendered );
+            $previousState.length ? $previousState.html( $rendered.html() ) : this.$el.append($rendered);
```
Comment (oliverfoster) [inline src/core/js/views/componentView.js]: can you switch to an if statement?

```js
var isFirstRun = !($previousState.length);
if (isFirstRun) {
  this.$el.append(rendered);
} else {
  $previousState.html(rendered);
}
```

File: src/core/js/views/componentView.js [hunk 2]
```diff
@@ -25,24 +25,18 @@ define([
...
 			// do not perform if component has .no-state class
             if (this.$el.is(".no-state")) return;
 
-			//remove pre-exisiting states
-            var $previousState = this.$(".accessibility-state").remove();
+            var $previousState = this.$(".accessibility-state");
 
-            //render and append state partial
-            var $rendered = $(Handlebars.partials['state']( this.model.toJSON() ));
+            var $rendered = $(Handlebars.partials['state']( _.extend(this.model.toJSON(), {a11yConfig:Adapt.config.get('_accessibility')}) ));
```
Comment (oliverfoster) [inline src/core/js/views/componentView.js]: multiple lines would be good
```js
var template = Handlebars.partials['state'];
var data =  _.extend(this.model.toJSON(), {
    a11yConfig: Adapt.config.get('_accessibility')
};
var rendered = template(data);
```
i don't think we need ``rendered`` as a jquery object and dom nodes, as in both cases we can use the raw html

**Comment by oliverfoster** [comment]
we're going to need a schema amend for the components on this one. @moloko could you take a look?

File: src/core/js/views/componentView.js (hunk 1 above)
Comment (chris-steele) [reply src/core/js/views/componentView.js]: Done

File: src/core/js/views/componentView.js (hunk 2 above)
Comment (chris-steele) [reply src/core/js/views/componentView.js]: Done

**Comment by moloko** [comment]
ah sorry, I should have read the original issue which contains an example ;-)

How about "Disables the hidden label that describes the state of the component to screenreader users. Useful if the component is of no interest to a screenreader user - such as an optional decorative graphic component."

**Comment by moloko** [comment]
And just to flag that the schema change will need to be done on the authoring tool repo as well...

**Pull request merged**